platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/FillManager.kt

```diff
@@ -0,0 +1,217 @@
+        style,
+        FillElementProvider(),
+        belowLayerId,
+        aboveLayerId,
+        geoJsonOptions
+    )
+
+    override fun initializeDataDrivenPropertyMap() = listOf(
+        FillOptions.PROPERTY_FILL_OPACITY,
+        FillOptions.PROPERTY_FILL_COLOR,
+        FillOptions.PROPERTY_FILL_OUTLINE_COLOR,
+        FillOptions.PROPERTY_FILL_PATTERN
+    ).associateWith { false }.let { dataDrivenPropertyUsageMap.putAll(it) }
+
+    override fun setDataDrivenPropertyIsUsed(property: String) {
+        when (property) {
+            FillOptions.PROPERTY_FILL_OPACITY -> layer.setProperties(
+                PropertyFactory.fillOpacity(
+                    Expression.get(FillOptions.PROPERTY_FILL_OPACITY)
+                )
+            )
+
+            FillOptions.PROPERTY_FILL_COLOR -> layer.setProperties(
+                PropertyFactory.fillColor(
+                    Expression.get(FillOptions.PROPERTY_FILL_COLOR)
+                )
+            )
+
+            FillOptions.PROPERTY_FILL_OUTLINE_COLOR -> layer.setProperties(
+                PropertyFactory.fillOutlineColor(
+                    Expression.get(FillOptions.PROPERTY_FILL_OUTLINE_COLOR)
+                )
+            )
+
+            FillOptions.PROPERTY_FILL_PATTERN -> layer.setProperties(
+                PropertyFactory.fillPattern(
+                    Expression.get(FillOptions.PROPERTY_FILL_PATTERN)
+                )
+            )
+        }
+    }
+
+    /**
+     * Create a list of fills on the map.
+     *
+     *
+     * Fills are going to be created only for features with a matching geometry.
+     *
+     *
+     * All supported properties are:<br></br>
+     * FillOptions.PROPERTY_FILL_OPACITY - Float<br></br>
+     * FillOptions.PROPERTY_FILL_COLOR - String<br></br>
+     * FillOptions.PROPERTY_FILL_OUTLINE_COLOR - String<br></br>
+     * FillOptions.PROPERTY_FILL_PATTERN - String<br></br>
+     * Learn more about above properties in the [Style specification](https://www.mapbox.com/mapbox-gl-js/style-spec/).
+     *
+     *
+     * Out of spec properties:<br></br>
+     * "is-draggable" - Boolean, true if the fill should be draggable, false otherwise
+     *
+     * @param json the GeoJSON defining the list of fills to build
+     * @return the list of built fills
+     */
+    @UiThread
+    fun create(json: String): List<Fill> = create(FeatureCollection.fromJson(json))
+
+    /**
+     * Create a list of fills on the map.
+     *
+     *
+     * Fills are going to be created only for features with a matching geometry.
+     *
+     *
+     * All supported properties are:<br></br>
+     * FillOptions.PROPERTY_FILL_OPACITY - Float<br></br>
+     * FillOptions.PROPERTY_FILL_COLOR - String<br></br>
+     * FillOptions.PROPERTY_FILL_OUTLINE_COLOR - String<br></br>
+     * FillOptions.PROPERTY_FILL_PATTERN - String<br></br>
+     * Learn more about above properties in the [Style specification](https://www.mapbox.com/mapbox-gl-js/style-spec/).
```

Comment:
Let's replace there while we are at it
```suggestion
     * Learn more about above properties in the [Style specification](https://maplibre.org/maplibre-style-spec/).
```